PWGJE/Tasks/jetCrossSectionEfficiency.cxx

```diff
@@ -0,0 +1,554 @@
+      }
+      histMcCollisionsEventSelectionWeighted->GetXaxis()->SetBinLabel(binZvtx, "zvtx");
+      histMcCollisionsEventSelectionWeighted->GetXaxis()->SetBinLabel(binCentrality, "centralitycut");
+      histMcCollisionsEventSelectionWeighted->GetXaxis()->SetBinLabel(binOccupancy, "occupancycut");
+    }
+  }
+
+  template <typename TTracks, typename TJets>
+  bool isAcceptedJet(TJets const& jet)
+  {
+    if (jetAreaFractionMin > configSwitchLow) {
+      if (jet.area() < jetAreaFractionMin * o2::constants::math::PI * (jet.r() / 100.0) * (jet.r() / 100.0)) {
+        return false;
+      }
+    }
+    bool checkConstituentMinPt = (leadingConstituentPtMinMCP > configSwitchLow);
+    bool checkConstituentMaxPt = (leadingConstituentPtMaxMCP < configSwitchHigh);
+    bool checkConstituentPt = checkConstituentMinPt || checkConstituentMaxPt;
+
+    if (checkConstituentPt) {
+      bool isMinLeadingConstituent = !checkConstituentMinPt;
+      bool isMaxLeadingConstituent = true;
+
+      for (const auto& constituent : jet.template tracks_as<TTracks>()) {
+        double constituentPt = constituent.pt();
+
+        if (checkConstituentMinPt && constituentPt >= leadingConstituentPtMinMCP) {
+          isMinLeadingConstituent = true;
+        }
+        if (checkConstituentMaxPt && constituentPt > leadingConstituentPtMaxMCP) {
+          isMaxLeadingConstituent = false;
+        }
+      }
+      return isMinLeadingConstituent && isMaxLeadingConstituent;
+    }
+    return true;
+  }
+
+  void processCrossSectionEfficiency(aod::JetMcCollisions::iterator const& mccollision,
+                                     soa::SmallGroups<aod::JetCollisionsMCD> const& collisions,
+                                     soa::Join<aod::ChargedMCParticleLevelJets, aod::ChargedMCParticleLevelJetConstituents> const& jets,
+                                     aod::JetParticles const&)
+  {
+    bool hasRecoColl = (collisions.size() >= 1);
+    bool passesZvtxCut = (std::abs(mccollision.posZ()) <= vertexZCut);
+    bool passesSplitCollCut = !(acceptSplitCollisions == NonSplitOnly && collisions.size() > 1);
+
+    std::vector<bool> stagePassed(activeStages.size(), false);
+    bool hasCustomEventSel = false;
+    bool centralityIsGood = false;
+    bool occupancyIsGood = false;
+    float centrality = mccollision.centFT0M();
+    if (acceptSplitCollisions == SplitOkCheckFirstAssocCollOnly) {
+      if (hasRecoColl) {
+        auto const& collision = collisions.begin();
+        for (std::size_t i = 0; i < activeStages.size(); ++i) {
+          if (collision.eventSel() & (1 << activeStages[i].bit)) {
+            stagePassed[i] = true;
+          }
+        }
+        if (jetderiveddatautilities::selectCollision(collision, eventSelectionBits, skipMBGapEvents)) {
+          hasCustomEventSel = true;
+        }
+        if ((trackOccupancyInTimeRangeMin < collision.trackOccupancyInTimeRange()) && (collision.trackOccupancyInTimeRange() < trackOccupancyInTimeRangeMax)) {
```

Comment:
these could be filters in your task I think? Also isnt this for pp? Why do you need occupancy?